### ISSUE TITLE
Wildcard versioning not working correctly

### ISSUE THREAD

**Issue by DennisReddyhoff** [FAIRDataPipeline/FAIR-CLI#44]
When using wildcards, the version is not correctly returning as '0.0.1'

**Comment by kzscisoft** [comment]
@DennisReddyhoff Any details on this? i.e. inputs etc

**Comment by richardreeve** [comment]
I understand that when you run a user config with a wildcard in the `write` block, like `test/*` with version `${{ PATCH }}`, then if it matches one file, say version `0.0.2` of `test/first`, then when it's writing out the working config, it tells you to use version `0.0.2` of any new wildcard match `test/*`, like `test/second` instead of counting from `0.0.0` and giving you `0.0.1` like it should. Does that make sense? I'm sure we can make an example if not!

**Comment by kzscisoft** [comment]
So you are saying that the version variables are working but just not for wildcard statements? This could be an ordering thing, maybe I put the wildcard parsing AFTER the version parsing when it should come before. 

The versioning should be working for most cases

**Comment by kzscisoft** [comment]
Hmm it should be working, I could do with a failing example

**Comment by richardreeve** [comment]
Okay. Here you go... I've put three files in Zulip as I can't upload them here. First [register.yaml](https://zulip.scrc.uk/user_uploads/2/2f/Ojh6yQuSW-tOoMzqdnPPLJk-/register.yaml) then [write.yaml](https://zulip.scrc.uk/user_uploads/2/c6/u1MzeF1NA6u8jHbULpPfqlYP/write.yaml) produces [config.yaml](https://zulip.scrc.uk/user_uploads/2/e9/8kSWgro1EM0fZm9C1eRsmpiP/config.yaml). This is just Sonia's basic registration example:

```sh
(fair-6UDMIGpx-py3.8) (base) bleaberry[~/Documents/git/julia/EcoSISTEM] fair pull register.yaml
Updating registry from config.yaml
(fair-6UDMIGpx-py3.8) (base) bleaberry[~/Documents/git/julia/EcoSISTEM] fair pull write.yaml
Updating registry from config.yaml
(fair-6UDMIGpx-py3.8) (base) bleaberry[~/Documents/git/julia/EcoSISTEM] fair run write.yaml
Updating registry from config.yaml
Hello World!
```

`register.yaml` just creates a data product called `records/SARS-CoV-2/scotland/human-mortality` with version `0.20210901.0`. `write.yaml` then writes to the `records/SARS-CoV-2/scotland/*`.  Then `config.yaml` then has:
```yaml
write:
- data_product: records/SARS-CoV-2/scotland/*
  file_type: csv
  use:
    version: 0.20210901.1
- data_product: records/SARS-CoV-2/scotland/human-mortality
  file_type: csv
  use:
    version: 0.20210901.1
```
This is the right version for `records/SARS-CoV-2/scotland/human-mortality`, but `records/SARS-CoV-2/scotland/*` should just be `0.0.1`.

**Comment by richardreeve** [comment]
Closed in #57.
